kyaml/kio/kio.go

```diff
@@ -138,8 +136,8 @@ func (p Pipeline) ExecuteWithCallback(callback PipelineExecuteCallbackFunc) erro
 
 		// If either the internal annotations for path, index, and id OR the legacy
 		// annotations for path, index, and id are changed, we have to update the other.
-		err = reconcileInternalAnnotations(result, nodeAnnos)
-		if err != nil && err != yaml.ErrMissingMetadata {
+		err = reconcileInternalAnnotations(result, nodeAnnos, false)
```

Comment:
what is the reason for providing `false` here? In what situation would we not want the output to match the input?

Reply:
Updated